File: sdk/elixir/examples/native_core.exs
# Read-only tour of the native Rocksky core (no auth needed).
#
#   ROCKSKY_ERL_PATH=../erlang mix run examples/native_core.exs
#
# Needs the native lib — build ../erlang/build-core.sh first.
# The write side (login + scrobble) is shown commented at the bottom.

{:ok, stats} = Rocksky.global_stats()
IO.puts("global: #{stats["scrobbles"]} scrobbles · #{stats["users"]} users · #{stats["tracks"]} tracks")

IO.puts("top tracks:")

{:ok, top} = Rocksky.top_tracks(5, 0)

for t <- top do
  IO.puts("  #{t["artist"]} — #{t["title"]}")
end

IO.puts("song hash: #{Rocksky.song_hash("Chaser", "Calibro 35", "Jazzploitation")}")

# --- write side (uncomment with real credentials) ---
# agent = Rocksky.login("session.json", "alice.bsky.social", "app-pw")
# {:ok, out} = Rocksky.scrobble(agent, %{
#   "title" => "Chaser", "artist" => "Calibro 35",
#   "album" => "Jazzploitation", "albumArtist" => "Calibro 35", "durationMs" => 182_320
# })
# IO.puts("scrobbled: #{out["scrobbleUri"]}")

File: sdk/ruby/examples/native_core.rb
# frozen_string_literal: true

# Read-only tour of the native Rocksky core (no auth needed).
#
#   ruby -Ilib examples/native_core.rb
#
# Needs the native lib — run ./build-core.sh first.
# The write side (login + scrobble) is shown commented at the bottom.
require "rocksky"

stats = Rocksky.global_stats
puts "global: #{stats["scrobbles"]} scrobbles · #{stats["users"]} users · #{stats["tracks"]} tracks"

puts "top tracks:"
Rocksky.top_tracks(limit: 5).each do |t|
  puts "  #{t["artist"]} — #{t["title"]}"
end

puts "song hash: #{Rocksky.song_hash("Chaser", "Calibro 35", "Jazzploitation")}"

# --- write side (uncomment with real credentials) ---
# agent = Rocksky::Agent.login("session.json", "alice.bsky.social", "app-pw")
# out = agent.scrobble(
#   "title" => "Chaser", "artist" => "Calibro 35",
#   "album" => "Jazzploitation", "albumArtist" => "Calibro 35", "durationMs" => 182_320
# )
# puts "scrobbled: #{out["scrobbleUri"]}"
# agent.close
